crates/sqlexec/src/engine.rs

```diff
@@ -30,30 +34,109 @@ pub struct SessionStorageConfig {
     pub gcs_bucket: Option<String>,
 }
 
+// TODO: There's a significant amount of overlap with `StorageConfig`, would be good to consider
+// consolidating them into one
 /// Storage configuration for the compute engine.
 ///
 /// The configuration defined here alongside the configuration passed in through
 /// the proxy will be used to connect to database storage.
 #[derive(Debug, Clone)]
 pub enum EngineStorageConfig {
```

Comment:
> If that was the case there would be no need for try_from_options logic, though some kind of validation logic would still be needed.

This should be fine. We do want something concrete after validation.

> Also needed would be the logic for hydrating the creds from the user's home if not supplied (the object store builders know how to read appropriate env vars on their own).

This would be a good addition!

Reply:
> This should be fine. We do want something concrete after validation.

Thanks. Just to clarify, by concrete do you mean a single map with all the validated options or enum variants with explicit struct-like fields (as is done currently)? Basically trying to determine which one you'd prefer.

Reply:
> Just to clarify, by concrete do you mean a single map with all the validated options or enum variants with explicit struct-like fields (as is done currently)? Basically trying to determine which one you'd prefer.

The latter (as current implementation).

Reply:
Got it, will leave that part as is then, thanks!